ArduPlane/events.cpp

```diff
@@ -146,13 +146,38 @@ void Plane::low_battery_event(void)
     if (flight_stage != AP_SpdHgtControl::FLIGHT_LAND_FINAL &&
         flight_stage != AP_SpdHgtControl::FLIGHT_LAND_PREFLARE &&
         flight_stage != AP_SpdHgtControl::FLIGHT_LAND_APPROACH) {
-    	set_mode(RTL, MODE_REASON_BATTERY_FAILSAFE);
-    	aparm.throttle_cruise.load();
+        set_mode(RTL, MODE_REASON_BATTERY_FAILSAFE);
+        aparm.throttle_cruise.load();
     }
     failsafe.low_battery = true;
     AP_Notify::flags.failsafe_battery = true;
 }
 
+void Plane::critical_battery_event(void)
+{
+    if (failsafe.critical_battery) {
+        return;
+    }
+    gcs_send_text_fmt(MAV_SEVERITY_WARNING, "Critical battery %.2fV used %.0f mAh",
+                      (double)battery.voltage(), (double)battery.current_total_mah());
+    if (flight_stage != AP_SpdHgtControl::FLIGHT_LAND_FINAL &&
+        flight_stage != AP_SpdHgtControl::FLIGHT_LAND_PREFLARE &&
+        flight_stage != AP_SpdHgtControl::FLIGHT_LAND_APPROACH) {
+        set_mode(RTL, MODE_REASON_BATTERY_FAILSAFE);
+        aparm.throttle_cruise.load();
+
+        // Go directly to the landing sequence
+        jump_to_landing_sequence();
```

Comment:
RTL_AUTOLAND 3 being this new behaviour would be fine. The objection is that none of the current RTL_AUTOLAND options allow RTL without also doing a landing. I want RTL on low battery, but I want to have time before just going into the landing sequence, and on normal RTL there should be a option that says loiter indefinitely at a rallypoint/home until battery is low, then land. That is primarily what this is meant to address. If you think RTL_AUTOLAND 3 meaning perform a normal RTL until critical battery and only then perform a autoland I'm fine with that.